pkg/resourcegroup/controllers/resourcegroup/resourcegroup_controller.go

```diff
@@ -346,23 +346,45 @@ func (r *reconciler) computeStatus(
 	return statuses
 }
 
-// ActuationStatusToLegacy contains the logic/rules to convert from the actuation statuses
-// to the legacy status field. If conversion is not needed, the original status field is returned
-// instead.
-func ActuationStatusToLegacy(s v1alpha1.ResourceStatus) v1alpha1.Status {
-	if s.Status == v1alpha1.NotFound {
-		return v1alpha1.NotFound
+// UpdateStatusToReflectActuation updates the latest computed kstatus to reflect
+// the desired state in the source of truth, as much as possible, not just the
+// desired state that has been persisted to the Kubernetes API.
+//
+// This is necessary, because kstatus only reflects whether the object status
+// reflects the object spec. But in Config Sync, the desired state is actually
+// in the source or truth. So if the latest desired state has not yet been
+// successfully actuated (applied or deleted), then the current status is
+// Unknown, because neither Kubernetes nor this ResourceGroup controller knows
+// what the desired state is.
+func UpdateStatusToReflectActuation(s v1alpha1.ResourceStatus) v1alpha1.Status {
+	// When actuation is unknown, return the latest computed status.
+	// This should only ever happen when upgrading from an old version.
+	// It means the applier was last run with code that predates the addition
+	// of the strategy, actuation, and reconcile fields.
+	if s.Actuation == "" {
```

Comment:
nit: Can we combine them given that they all return `s.Status`? Or at least combine the `if s.Actuation` ones

Reply:
I considered that and figured it was valuable to explain the reasoning in the long comments before each case. But if you think it's important, I can go combine them and make a larger single comment block with bullet points.